src/parameterizations/lateral/MOM_tidal_forcing.F90

```diff
@@ -362,18 +361,12 @@ subroutine tidal_forcing_init(Time, G, US, param_file, CS, HA_CS)
                  "If true, use the SAL from the previous iteration of the "//&
                  "tides to facilitate convergent iteration. "//&
                  "This is only used if TIDES is true.", default=.false.)
-  call get_param(param_file, '', "TIDE_SAL_SCALAR_VALUE", tide_sal_scalar_value, &
-                 units="m m-1", default=0.0, do_not_log=.True.)
-  if (tide_sal_scalar_value/=0.0) &
-    call MOM_error(WARNING, "TIDE_SAL_SCALAR_VALUE is a deprecated parameter. "//&
-                   "Use SAL_SCALAR_VALUE instead." )
-  call get_param(param_file, mdl, "SAL_SCALAR_VALUE", CS%sal_scalar, &
-                 "The constant of proportionality between sea surface "//&
-                 "height (really it should be bottom pressure) anomalies "//&
-                 "and bottom geopotential anomalies. This is only used if "//&
-                 "USE_SAL_SCALAR is true or USE_PREVIOUS_TIDES is true.", &
-                 default=tide_sal_scalar_value, units="m m-1", &
-                 do_not_log=(.not. CS%use_tidal_sal_prev))
+  if (CS%use_tidal_sal_prev) &
+    call get_param(param_file, mdl, "SAL_SCALAR_VALUE", CS%sal_scalar, "The constant of "//&
+                   "proportionality between self-attraction and loading (SAL) geopotential "//&
+                   "anomaly and barotropic geopotential anomalies. This is only used if "//&
+                   "SAL_SCALAR_APPROX is true or USE_PREVIOUS_TIDES is true.", default=0.0, &
+                   units="m m-1", do_not_log=.true., old_name='TIDE_SAL_SCALAR_VALUE')
```

Comment:
Please reset the `do_not_log` settings here to `do_not_log=(.not.CS%use_tidal_sal_prev)` here.  There are parameter settings for which `SAL_SCALAR_VALUE` would be set here, but `SAL_init()` would not have been called.  These settings do not make much sense, but I think that they could have been used, so `SAL_SCALAR_VALUE` should be logged in that case.  Also, a parameter is never logged more than once, so changing the `do_not_log` settings here should be benign in the usual (logically sensible) case.

Reply:
Good catch!